src/test-run/commands/index.js

```diff
@@ -1,25 +1,54 @@
-export const TYPE = {
-    click:     'click',
-    assertion: 'assertion',
-    testDone:  'test-done'
-};
-
-export function createClickCommand () {
-    return {
-        type:      TYPE.click,
-        arguments: {}
-    };
+import TYPE from './type';
+import Assignable from '../../utils/assignable';
+import { ActionSelectorError } from '../../errors/test-run';
+import { ClickOptions } from './options';
+
+// Validators
+function selector (option, val) {
+    if (typeof val !== 'string')
+        throw new ActionSelectorError(val);
 }
 
-export function createAssertionCommand () {
-    return {
-        type:      TYPE.assertion,
-        arguments: {}
-    };
+function clickOptions (option, val) {
+    return new ClickOptions(val, true);
```

Comment:
Yep, moreover, we will be able to validate this way, but we will not filter unnecessary properties and defaults.